db/conn.go

```diff
@@ -45,60 +69,67 @@ func GetURL(params connectionParams) string {
 	)
 }
 
-func init() {
-	initializeConnection()
+// GetDefaultDSN is a wrapper for GetDSN without params.
+func GetDefaultDSN() string {
+	return GetDSN(ConnParams{})
 }
 
-func openDefaultDB() {
-	var err error
-
-	dbURL := GetURL(connectionParams{})
-	monitor.Logger.WithFields(log.Fields{
-		"db_url": dbURL,
-	}).Info("connecting to the database")
-	Conn, err = gorm.Open("postgres", dbURL)
+// NewConnection sets up a database object, panics if unable to connect.
+func NewConnection(dsn string) *ConnData {
+	c := ConnData{dialect: "postgres", Logger: monitor.NewModuleLogger("db")}
+	c.Logger.LogF(monitor.F{"dsn": dsn}).Info("connecting to the database")
+	db, err := connect(c.dialect, dsn)
 	if err != nil {
 		panic(err)
 	}
+
+	c.DB = db
+	return &c
 }
 
-func initializeConnection() {
-	dbName := config.Settings.GetString("DatabaseName")
-	dbURL := GetURL(connectionParams{DatabaseName: "postgres"})
+func connect(dialect, dsn string) (*sqlx.DB, error) {
+	conn, err := sqlx.Connect(dialect, dsn)
+	return conn, err
+}
 
-	if dbName == "" {
-		panic("DatabaseName not configured")
+// MigrateUp executes forward migrations.
+func (c ConnData) MigrateUp() {
```

Comment:
Thought about it for some time and figured that both `ConnData` and `Logger` are shallow structures containing pointers to larger things, i.e. passing them around by value instead of pointer should be alright, as long as we're not modifying them.